### PR TITLE
INTEXT-208: The "weeding" around expressions

### PR DESCRIPTION
**Pull request by artembilan**
JIRA: https://jira.spring.io/browse/INTEXT-208

### CODE REVIEW

**Comment by garyrussell** [comment]
I am not sure what "weeding" means here, but this does not appear to address the JIRA issue (simplifying the use of expressions by providing `String` alternatives).

File: src/main/java/org/springframework/integration/dsl/http/HttpMessageHandlerSpec.java [hunk 1]
```diff
@@ -182,11 +182,55 @@ public HttpMessageHandlerSpec uriVariable(String variable, Expression value) {
...
 	}
 
+	/**
+	 * @param variable the uri template variable.
+	 * @param value the expression to evaluate value for te uri template variable.
+	 * @return the current Spec.
+	 * @see HttpRequestExecutingMessageHandler#setUriVariableExpressions(Map)
+	 * @since 1.1.1
+	 */
+	public HttpMessageHandlerSpec uriVariable(String variable, String value) {
```
Comment (artembilan) [inline src/main/java/org/springframework/integration/dsl/http/HttpMessageHandlerSpec.java]: I'm addressing JIRA here.

File: src/main/java/org/springframework/integration/dsl/http/HttpMessageHandlerSpec.java [hunk 2]
```diff
@@ -182,11 +182,55 @@ public HttpMessageHandlerSpec uriVariable(String variable, Expression value) {
...
+
+	/**
+	 * @param variable the uri template variable.
+	 * @param valueFunction the function to evaluate value for te uri template variable.
+	 * @param <P> the payload type.
+	 * @return the current Spec.
+	 * @see HttpRequestExecutingMessageHandler#setUriVariableExpressions(Map)
+	 * @since 1.1.1
+	 */
+	public <P> HttpMessageHandlerSpec uriVariable(String variable, Function<Message<P>, ?> valueFunction) {
```
Comment (artembilan) [inline src/main/java/org/springframework/integration/dsl/http/HttpMessageHandlerSpec.java]: And here

File: src/main/java/org/springframework/integration/dsl/http/HttpMessageHandlerSpec.java [hunk 3]
```diff
@@ -182,11 +182,55 @@ public HttpMessageHandlerSpec uriVariable(String variable, Expression value) {
...
 		return this;
 	}
 
+	/**
+	 * @param uriVariablesExpression to use.
+	 * @return the current Spec.
+	 * @see HttpRequestExecutingMessageHandler#setUriVariablesExpression(Expression)
+	 * @since 1.1.1
+	 */
+	public HttpMessageHandlerSpec uriVariablesExpression(String uriVariablesExpression) {
```
Comment (artembilan) [inline src/main/java/org/springframework/integration/dsl/http/HttpMessageHandlerSpec.java]: Plus here.
I dug the whole project and haven't found so much much vulnerabilities around `string expression representations`.

**Comment by garyrussell** [comment]
Doh; sorry - I just saw you adding a number of `Expression` methods :smile: Monday and no :coffee:

**Pull request merged**